include/deal.II/matrix_free/tools.h

```diff
@@ -793,16 +804,26 @@ namespace MatrixFreeTools
     const unsigned int                                              quad_no,
     const unsigned int first_selected_component)
   {
-    using VectorType = LinearAlgebra::distributed::Vector<Number>;
-
     // initialize vector
-    matrix_free.initialize_dof_vector(diagonal_global, dof_no);
+    if (IsBlockVector<VectorType>::value)
+      {
+        for (unsigned int comp = 0; comp < n_components; ++comp)
+          {
+            auto vec = ::dealii::internal::BlockVectorSelector<
+              VectorType,
+              IsBlockVector<VectorType>::value>::
+              get_vector_component(diagonal_global, comp);
+            matrix_free.initialize_dof_vector(*vec, dof_no);
+          }
+      }
+    else
+      matrix_free.initialize_dof_vector(diagonal_global, dof_no);
```

Comment:
This part is under discussion: should we initialize the vector inside this function or should we enforce the user to do this a priori?

Reply:
I think https://github.com/dealii/dealii/pull/13316 is my answer to this question ;)